draft-ietf-httpbis-origin-frame.md

```diff
@@ -101,49 +106,65 @@ REMOVE (0x2):
+released from the requirement to establish authority for a given origin using DNS, for that
+connection. However, they MUST still establish authority using the certificate, as described in
+{{!RFC7540}} Section 9.1.1.
 
-A client that implements this specification MUST NOT use a connection for a given origin unless that origin appears in the Origin Set for the connection, regardless of whether or not it believes that the connection is authoritative for that origin.
+Once such a frame is received, an implementing client MUST NOT use that connection for a given
+origin unless it appears in the connection's Origin Set. Implementing clients SHOULD use a
+connection for all origins which it is authoritative for, if they are included in its Origin Set.
 
 
-## Processing ORIGIN Frames
+## Processing ORIGIN Frames {#process}
 
 The ORIGIN frame is a non-critical extension to HTTP/2. Endpoints that do not support this frame
 can safely ignore it upon receipt.
 
-When received by a client, it can be used to inform HTTP/2 connection coalescing (see {{set}}), but
-does not relax the requirement there that the server is authoritative.
+When received by an implementing client, it is used to manipulate the Origin Set (see {{set}}).
 
-The origin frame MUST be sent on stream 0; an ORIGIN frame on any other stream is invalid and MUST be ignored. 
+The origin frame MUST be sent on stream 0; an ORIGIN frame on any other stream is invalid and MUST
+be ignored.
 
 The ORIGIN frame is processed hop-by-hop. An intermediary MUST NOT forward ORIGIN frames. Clients
 configured to use a proxy MUST ignore any ORIGIN frames received from it.
 
 The following algorithm illustrates how a client can handle received ORIGIN frames:
 
-1. If the client is configured to use a proxy, ignore the frame and stop processing.
+1. If the client is configured to use a proxy for the connection, ignore the frame and stop processing.
 2. If the frame occurs upon any stream except stream 0, ignore the frame and stop processing.
 3. If the CLEAR flag is set, remove all members from the Origin Set.
 4. For each Origin field `origin_raw` in the frame payload:
-   1. Parse `origin_raw` as an ASCII serialization of an origin ({{!RFC6454}}, Section 6.2) and let the result be `parsed_origin`.
+   1. Parse `origin_raw` as an ASCII serialization of an origin ({{!RFC6454}}, Section 6.2) and let the result be `parsed_origin`. If parsing fails, skip to the next `origin_raw`.
    2. If the REMOVE flag is set, remove any member of the Origin Set that is the same as `parsed_origin` (as per {{!RFC6454}}, Section 5), and continue to the next `parsed_origin`.
    3. Otherwise, add `parsed_origin` to the Origin Set.
 
 
 # Security Considerations
 
 Clients that blindly trust the ORIGIN frame's contents will be vulnerable to a large number of
-attacks; hence the reinforcement that this specification does not relax the requirement for server
-authority in {{?RFC7540}}, Section 10.1.
+attacks.
```

Comment:
Rather than simple removal, I would add:

> Though this document removes the requirement on clients to ensure that the DNS refers to the same IP address, it does not remove the requirement for certificate-based authentication.  A client MUST validate a server IP address.